### ISSUE TITLE
Wrong shape on a specific node into TFLITE 32 model generated

### ISSUE THREAD

**Issue by PetiteFleurPF** [PINTO0309/onnx2tf#638]
### Issue Type

Others

### OS

Linux

### onnx2tf version number

1.21.0

### onnx version number

1.15.0

### onnxruntime version number

1.17.1

### onnxsim (onnx_simplifier) version number

0.4.33

### tensorflow version number

2.16.1

### Download URL for ONNX

[model.onnx.zip](https://github.com/PINTO0309/onnx2tf/files/15413105/model.onnx.zip)

### Parameter Replacement JSON

```json
{}
```

### Description

1. Purpose - Product development: The impact of the resolution of this bug will increase the speed of the models.
2.  What: At node 282 into TFLITE 32 converted model (observable number during the model conversation), a bug appears preventing the prediction from being obtained during the inference process. This problem can be summed up as a shape problem: ‘Given shapes, [100, 4] and [99, 1], are not broadcastable. Node number 276 (ADD) failed to prepare.’

INFO: 283 / 301
INFO: onnx_op_type: Gather onnx_op_name: /Gather_8
INFO:  input_name.1: /GatherND_1_output_0 shape: ['unk__211', 4] dtype: float32
INFO:  input_name.2: /TopK_output_1 shape: ['unk__327'] dtype: int64
INFO:  output_name.1: /Gather_8_output_0 shape: ['unk__327', 4] dtype: float32
INFO: tf_op_type: gather_v2
INFO:  input.1.params: name: tf.compat.v1.gather_nd_1/GatherNd:0 shape: (None, 4) dtype: <dtype: 'float32'> 
INFO:  input.2.indices: name: tf.__operators__.add_30/AddV2:0 shape: (None,) dtype: <dtype: 'int64'> 
INFO:  input.3.axis: val: 0 
INFO:  output.1.output: name: tf.compat.v1.gather_3/GatherV2:0 shape: (None, 4) dtype: <dtype: 'float32'> 

3. How: I inspected the model to identify the precise step. I used the onnx2tf tool to identify the match and unmatch’ - I was then able to see that there was a slight difference in accuracy in one of the two inputs. However, this difference disappears when using the -cotoa 1e-1 option. The other input will have skipped values which are removed by a simple change in the code to accept dimensions equivalent to 1. I noticed that all the reshaping steps were unmatched before the option.
4. Why: Because I really need to improve the inference time.

**Comment by AmelieNef** [comment]
![image](https://github.com/PINTO0309/onnx2tf/assets/45994072/16c17e4d-e771-480f-a710-aa00ab1376e5)
current schema of the model to the node (in bold at the bottom) which is generating the error.

**Comment by PINTO0309** [comment]
I will not investigate unless you share the ONNX files with me.

**Comment by PetiteFleurPF** [comment]
@PINTO0309 Hi PINTO0309 - I updated my description with a link to download my model :)

**Comment by PINTO0309** [comment]
Post-processing is too redundant. There are no shapes that can be estimated by the tool after `NonZero`. Models in which the number of output elements varies with the content of the input data cannot be accurately transformed by onnx2tf. This is not a bug.

![image](https://github.com/PINTO0309/onnx2tf/assets/33194443/7fb4b14d-5899-4cf4-9c9b-cdaad2c54177)

![image](https://github.com/PINTO0309/onnx2tf/assets/33194443/68680e81-7393-46af-a124-f68b1446d97e)

Cut out the model above the red line.

![image](https://github.com/PINTO0309/onnx2tf/assets/33194443/6dfa7c1b-6a12-459e-99ac-9b2007a61e7b)

![image](https://github.com/PINTO0309/onnx2tf/assets/33194443/bf8110f9-f894-48e2-857d-0962ea763df4)

**Comment by PetiteFleurPF** [comment]
OK, let me rephrase - as I couldn't legally send you the model I was working on, I sent you one that reproduced the error once the conversion had been done - and which passed any inference. As mentioned above, the validation tests were all OK except for those shown in the screen. Theoretically I can't modify my model, so if I understand correctly the solution is not to modify your code but my model?

**Comment by PINTO0309** [comment]
> if I understand correctly the solution is not to modify your code but my model?

Yes. Aside from the fact that it can and cannot be converted, the model is too redundant and inefficient. In particular, all models generated without modification of `Detectron` and others will generate this kind of redundant post-processing.

The post-processing of your model can be replaced by the process shown below.

![image](https://github.com/PINTO0309/onnx2tf/assets/33194443/e871b463-5f2f-4c57-8e13-b08248d669b0)

**Comment by PetiteFleurPF** [comment]
OK many thanks :)

**Comment by PetiteFleurPF** [comment]
Question - what tool do you recommend to do it, please?  ONNX surgeon?

**Comment by PINTO0309** [comment]
https://github.com/PINTO0309/PINTO_model_zoo/tree/main/307_YOLOv7
https://github.com/PINTO0309/PINTO_model_zoo/tree/main/322_YOLOv7_Head
https://github.com/PINTO0309/PINTO_model_zoo/tree/main/326_YOLOPv2
https://github.com/PINTO0309/PINTO_model_zoo/tree/main/334_DAMO-YOLO
https://github.com/PINTO0309/PINTO_model_zoo/tree/main/337_FreeYOLO
https://github.com/PINTO0309/PINTO_model_zoo/tree/main/356_EdgeYOLO
https://github.com/PINTO0309/PINTO_model_zoo/tree/main/447_YOLOX-Wholebody-with-Wheelchair

**Comment by PetiteFleurPF** [comment]
What I meant was a tool that lets me modify my model to change, replace or delete parts of the post-processing ^^ But that's OK, no worries.
